students/rob_sanchez/Lesson 3/mailroom.py

```diff
@@ -0,0 +1,166 @@
+                  {"donation-number": '2', "amount": '200'},
+                  {"donation-number": '3', "amount": '300'}
+                ]},
+              "Michael Jordan": {
+                "donations": [
+                    {"donation-number": '1', "amount": '1300'}
+                ]},
+              "Katy Perry": {
+                "donations": [
+                    {"donation-number": '1', "amount": '4500'},
+                    {"donation-number": '2', "amount": '1500'}
+                ]},
+              "Adam Sandler": {
+                "donations": [
+                    {"donation-number": '1', "amount": '500'},
+                    {"donation-number": '2', "amount": '2400'}
+                ]}
+              }
+
+
+def main():
+    print("\nWelcome!\n")
+    welcome()
+
+
+# Displays the initial welcome prompt
+def welcome():
+
+    # Prompt for the option to be executed
+    print("\nSelect an option below:\n")
+
+    selection = welcome_options()
+
+    if selection == '1':
+        send_thank_you()
+    elif selection == '2':
+        create_report()
+    elif selection == '3':
+        sys.exit()
+
+
+# Sends a thank you email to the selected donor
+def send_thank_you():
+    # Name of donor to send thank you email
+    donor_name = input("\nPlease enter the Donor's full name:\n")
+
+    # Create list of names
+    name_set = set()
+    for donor, value in donor_dict.items():
+        name_set.add(donor)
+
+    while donor_name.lower() == "list":
+        print ("\nList of Donors: " + print_list(list(name_set)))
+        donor_name = input("\nPlease enter the Donor's full name:\n")
+
+    if donor_name not in list(name_set):
+        amt_input = input("\nDonation amount:\n")
+
+        donor_dict[donor_name] = {"donations": [{"donation-number": '1', "amount": '0'}]}
+        donor_dict[donor_name]["donations"][0]["amount"] = float(amt_input)
+        send_email(donor_name, amt_input)
+    else:
+        amt_input = input("\nDonation amount:\n")
+        history = donor_dict[donor_name]["donations"]
+        num_of_donations = []
+
+        for item in history:
+            num_of_donations.append(item.get('amount'))
+
+        new_index = len(num_of_donations)
+
+        donor_dict[donor_name]["donations"].append(
+                            {"donation-number": '', "amount": ''})
+        donor_dict[donor_name]["donations"][new_index]["donation-number"] = new_index + 1
+        donor_dict[donor_name]["donations"][new_index]["amount"] = float(amt_input)
+
+        send_email(donor_name, amt_input)
+
+
+# Creates a summary report of the donations
+def create_report():
+    headers = ["Donor Name", "Total Given", "Num Gifts", "Average Gift"]
+    print(("{:<30} | {:^15} | {:^15} | {:^15}").format(*headers))
+    print("-"*82)
+
+    donor_report = get_summary()
+
+    for item in donor_report:
+        print(("{:<30} ${:>17} {:>16} ${:>14}").format(*item))
+
+    welcome()
```

Comment:
it is best to use `while True` loop instead of calling `welcome()` in each function.

Imagine you expand your program and having to remember to call `welcome` function to control the main program flow.

this shouldn't be the concern of "feature" functions.